Cache/Redis.php

```diff
@@ -35,13 +36,13 @@ class Redis implements Cache
     /**
      * Constructor
      *
-     * @param Predis|\Redis   $redisClient Redis client instance
-     * @param LoggerInterface $logger      Logger instance
-     * @param string          $keyPrefix   A prefix to append
+     * @param PredisClient|RedisClient $redisClient Redis client instance
+     * @param LoggerInterface          $logger      Logger instance
+     * @param string                   $keyPrefix   A prefix to append
      */
     public function __construct($redisClient, LoggerInterface $logger = null, $keyPrefix = self::KEY_PREFIX_DEFAULT)
     {
-        if (!$redisClient instanceof Predis && !(class_exists('Redis') && $redisClient instanceof \Redis)) {
+        if (!$redisClient instanceof PredisClient && !$redisClient instanceof RedisClient) {
```

Comment:
this was overcomplicated. instanceof for a not existing class is not a problem in php, see https://3v4l.org/OG7tG